### PR TITLE
Fix remaining notebooks in dataset folder

### PR DESCRIPTION
**Pull request by lakhotiaharshit**
Some minor fixes in notebooks 
- DO ND using the dataset
- Linking to parent datasets
- The experiment container
- Working with snapshots
- Working with pandas and xarrays

To do:

- [x] Fix all broken links after renaming the dataset context manager notebook

@GateBuilder @astafan8 @WilliamHPNielsen

### CODE REVIEW

**Comment by jenshnielsen** [comment]
If we are renaming the dataset context manager notebook I suggest that we call it something completely different more along the line of performing  measurements with the qcodes dataset and change some of the prose in the notebook too to reflect that

File: docs/examples/DataSet/Implementing_doND_using_the_dataset.ipynb [hunk 1]
```diff
@@ -9,41 +9,57 @@
...
+     "execution_count": 10,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "initialise_database()  # just in case no database file exists\n",
-    "new_experiment(\"doNd-tutorial\", sample_name=\"no sample\")"
+    "tutorial_db_path = os.path.join(os.getcwd(), 'tutorial_doND.db')\n",
+    "initialise_or_create_database_at(tutorial_db_path)  # just in case no database file exists\n",
```
Comment (astafan8) [inline docs/examples/DataSet/Implementing_doND_using_the_dataset.ipynb]:
```suggestion
    "initialise_or_create_database_at(tutorial_db_path)\n",
```

File: docs/examples/DataSet/Linking to parent datasets.ipynb [hunk 2]
```diff
@@ -159,22 +159,22 @@
...
      },
-     "execution_count": 6,
+     "execution_count": 22,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "raw_data = load_by_id(raw_run_id)\n",
+    "raw_data = load_by_run_spec(captured_run_id = dataset.captured_run_id)\n",
```
Comment (astafan8) [inline docs/examples/DataSet/Linking to parent datasets.ipynb]: readability vs pep8 :)
```suggestion
    "raw_data = load_by_run_spec(captured_run_id=dataset.captured_run_id)\n",
```

File: docs/examples/DataSet/The-Experiment-Container.ipynb [hunk 3]
```diff
@@ -9,113 +9,41 @@
...
-    "# now you could save the config file in home. \n",
-    "# NOTE: This config file will \n",
-    "# then be used from now on loaded by \n",
-    "# qcodes instead of the default.\n",
-    "\n",
-    "# configuration.save_to_home()\n",
-    "# create an empty database based on the config file\n",
-    "initialise_database()"
+    "db_file_path = os.path.join(os.getcwd(),'exp_container_tutorial.db')\n",
+    "initialise_or_create_database_at(db_file_path )"
```
Comment (astafan8) [inline docs/examples/DataSet/The-Experiment-Container.ipynb]:
```suggestion
    "initialise_or_create_database_at(db_file_path)"
```

File: docs/examples/DataSet/The-Experiment-Container.ipynb [hunk 4]
```diff
@@ -9,113 +9,41 @@
...
-    "configuration['core']['db_location'] = './exp_container_tutorial.db'\n",
-    "# now you could save the config file in home. \n",
-    "# NOTE: This config file will \n",
-    "# then be used from now on loaded by \n",
-    "# qcodes instead of the default.\n",
-    "\n",
-    "# configuration.save_to_home()\n",
-    "# create an empty database based on the config file\n",
-    "initialise_database()"
+    "db_file_path = os.path.join(os.getcwd(),'exp_container_tutorial.db')\n",
```
Comment (astafan8) [inline docs/examples/DataSet/The-Experiment-Container.ipynb]:
```suggestion
    "db_file_path = os.path.join(os.getcwd(), 'exp_container_tutorial.db')\n",
```

File: docs/examples/DataSet/The-Experiment-Container.ipynb [hunk 5]
```diff
@@ -231,12 +159,12 @@
    "source": [
     "We notice that each experiment is labelled by an integer. This is the **exp_id** that can be used when looking up properties of each experiment.\n",
     "\n",
-    "Let us add some runs to experiment 2 (\"second_exp\"). For the sake of clarity, we don't add any data to the runs here. We refer to the `Dataset Context Manager.ipynb` for detailed information on how to properly create and populate runs."
+    "Let us add some runs to experiment 2 (\"second_exp\"). For the sake of clarity, we don't add any data to the runs here. We refer to the [Dataset Context Manager](Dataset-Context-Manager.ipynb) for detailed information on how to properly create and populate runs. We also refer to the [DataSet-class-walkthrough](DataSet-class-walkthrough.ipynb) notebook for more details on notebook."
```
Comment (astafan8) [inline docs/examples/DataSet/The-Experiment-Container.ipynb]:
```suggestion
    "Let us add some runs to experiment 2 (\"second_exp\"). For the sake of clarity, we don't add any data to the runs here. We refer to the [Dataset Context Manager](Dataset-Context-Manager.ipynb) for detailed information on how to properly create and populate runs. We also refer to the [DataSet-class-walkthrough](DataSet-class-walkthrough.ipynb) notebook for more details on DataSet class."
```

File: docs/examples/DataSet/The-Experiment-Container.ipynb [hunk 6]
```diff
@@ -338,13 +266,6 @@
    "source": [
     "In a similar way, we may of course add runs to the other two experiments. Now, the alert reader will have noticed that `exp_id` is a keyword argument in `new_data_set`. What is the default then? --The default is that the run is added to the experiment with the **highest** `exp_id` in the database."
```
Comment (astafan8) [inline docs/examples/DataSet/The-Experiment-Container.ipynb]: could you still please add a note that new_data_set here is used for the sake of excersise and should not be used in "real life"? :)

Comment (lakhotiaharshit) [reply docs/examples/DataSet/The-Experiment-Container.ipynb]: Done

**Comment by jenshnielsen** [comment]
Looks good. Could you change ``The power of the new construct```  and the line below to not talk about a new construct but perhaps just say the power of the context manager construct or some such thing so we talk less about it being new (which is not really relevant anymore )

**Comment by astafan8** [comment]
@jenshnielsen did it.

**Pull request merged**